grafana/provider.go

```diff
@@ -58,10 +68,46 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	} else {
 		cfg.APIKey = auth[0]
 	}
+
+	headersObj := d.Get("headers").(map[string]interface{})
+	if headersObj != nil && len(headersObj) == 0 {
+		// Workaround for a bug when DefaultFunc returns a TypeMap
+		headersObjAbs, _ := JsonEnvDefaultFunc("GRAFANA_HTTP_HEADERS", nil)()
```

Comment:
Why are you deciding to ignore the error here? If `GRAFANA_HTTP_HEADERS` contains invalid JSON no error will be reported to the user but the provider will continue to work, so the end user won't know that their configuration was ignored.

Reply:
You are right, returning the error in this case.